sql/connect-api/pom.xml

```diff
@@ -0,0 +1,312 @@
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.gson</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.gson</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>javax.annotation</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.javax.annotation</shadedPattern>
+                        </relocation>
+                        <!--
+                          For `com.google.api.grpc:proto-google-common-protos`, do not directly define pattern
+                          as `common.google`, otherwise, the relocation result may be uncertain due
+                          to the change of rule order.
+                        -->
+                        <relocation>
+                            <pattern>com.google.api</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.api</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.cloud</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.cloud</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.geo</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.geo</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.logging</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.logging</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.longrunning</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.longrunning</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.rpc</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.rpc</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.google.type</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.google_protos.type</shadedPattern>
+                        </relocation>
+                    </relocations>
+                    <filters>
+                        <filter>
+                            <artifact>org.apache.tomcat:annotations-api</artifact>
+                            <includes>
+                                <include>javax/annotation/**</include>
```

Comment:
We actually only need javax.annotation.Generated. We are dropping the rest because shading all of javax is not something we want to do. The alternative is that we don't shade `org.apache.tomcat:annotations-api` and `com.google.code.findbugs:jsr305`.